### PR TITLE
Debounce filter & contingency creation, useNameField

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/components/dialogs/create-contingency-list-dialog.js [hunk 1]
```diff
@@ -98,58 +98,60 @@ export const CreateContingencyListDialog = ({ open, onClose }) => {
...
-        } else {
-            setContingencyFormState('', false);
-            setLoadingCheckContingencyName(false);
-        }
-    };
+        },
+        [activeDirectory, intl]
+    );
 
+    const debouncedUpdateContingencyFormState = useMemo(
```
Comment (sBouzols) [inline src/components/dialogs/create-contingency-list-dialog.js]: useCallback

File: src/components/dialogs/create-filter-dialog.js [hunk 2]
```diff
@@ -136,54 +135,57 @@ const CreateFilterDialog = ({
...
-        } else {
-            setFilterFormState('', false);
-            setLoadingCheckFilterName(false);
-        }
-    };
+        },
+        [activeDirectory, intl]
+    );
+
+    const debouncedUpdateFilterFormState = useMemo(
```
Comment (sBouzols) [inline src/components/dialogs/create-filter-dialog.js]: useCallback

File: src/components/dialogs/field-hook.js [hunk 3]
```diff
@@ -237,6 +231,11 @@ export const useNameField = ({
         ]
     );
 
+    const debouncedUpdateValidity = useMemo(
```
Comment (sBouzols) [inline src/components/dialogs/field-hook.js]: useCallback

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/dialogs/create-contingency-list-dialog.js (hunk 1 above)
Comment (sBouzols) [reply src/components/dialogs/create-contingency-list-dialog.js]: nope useMemo is better you are right

File: src/components/dialogs/create-filter-dialog.js (hunk 2 above)
Comment (sBouzols) [reply src/components/dialogs/create-filter-dialog.js]: nope useMemo is better you are right

File: src/components/dialogs/field-hook.js (hunk 3 above)
Comment (sBouzols) [reply src/components/dialogs/field-hook.js]: nope useMemo is better you are right

**Comment by sBouzols** [comment]
Move your custom hook in commons-ui then use it here and all will be good :)
Same naming convention could be nice : `debounced...`

**Review by jonenst — APPROVED** [review]

**Review by sBouzols — APPROVED** [review]

**Pull request merged**